test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -491,6 +491,133 @@ public async Task OutputsValidJSONLogs()
+                enableExtendedSessions: false,
+                storageProviderType: "azure_storage"))
+            {
+                await host.StartAsync();
+
+                // This orchestrator should error out on null inputs
+                var client = await host.StartOrchestratorAsync(orchestratorName, input: null, this.output);
+                var status = await client.WaitForCompletionAsync(this.output);
+                await host.StopAsync();
+            }
+
+            // Ensure the logging file was at least generated
+            Assert.True(File.Exists(LinuxAppServiceLogger.LoggingPath));
+
+            // Ensure newlines are removed by checking the number of lines is equal to the number of "TimeStamp" columns,
+            // which corresponds to the number of JSONs logged
+            string[] lines = File.ReadAllLines(LinuxAppServiceLogger.LoggingPath);
+
+            bool foundAzureStorageLog = false;
+            bool foundEtwEventSourceLog = false;
+            bool foundDurableTaskCoreLog = false;
+
+            bool foundActivityId = false;
+            bool foundRelatedActivityId = false;
+
+            // Validating JSON outputs
+            foreach (string line in lines)
+            {
+                // (1) Ensure it can be parsed to JSON
+                JObject json = JObject.Parse(line);
+
+                // (2) Contains minimal expected fields, and does not contain
+                // deliberately missing fields
+                List<string> keys = json.Properties().Select(p => p.Name).ToList();
+                Assert.DoesNotContain("EventStampName", keys);
+                Assert.DoesNotContain("EventPrimaryStampName", keys);
+                Assert.DoesNotContain("RoleInstance", keys);
+                Assert.DoesNotContain("Tenant", keys);
+                Assert.Contains("ProviderName", keys);
+                Assert.Contains("TaskName", keys);
+                Assert.Contains("EventId", keys);
+                Assert.Contains("TimeStamp", keys);
+                Assert.Contains("Pid", keys);
+                Assert.Contains("Tid", keys);
+
+                // recording EventSource providers seen
+                int eventId = (int)json.GetValue("EventId");
+                if (eventId == 202)
+                {
+                    foundEtwEventSourceLog = true;
+                }
+                else if (eventId == 10)
+                {
+                    foundDurableTaskCoreLog = true;
+                }
+                else if (eventId == 120)
+                {
+                    foundAzureStorageLog = true;
+                }
+
+                // recording if ActivityId and RelatedActivityId are seen
+                // we expect to see them, at some point, in a trivial orchestrator
+                if (keys.Contains("ActivityId"))
+                {
+                    foundActivityId = true;
+                }
+
+                if (keys.Contains("ActivityId"))

```

Comment:
This is still wrong, should be RelatedActivityId. Not sure why that change keeps on getting removed

Reply:
done!